api/ws/rpc/connect.js

```diff
@@ -122,14 +122,18 @@ const connectSteps = {
 		// Register RPC methods on peer
 		peer = _.reduce(
 			wsServer.endpoints.rpc,
-			(peerExtendedWithRPC, localHandler, rpcProcedureName) => {
+			(peerExtendedWithRPC, _localHandler, rpcProcedureName) => {
 				peerExtendedWithRPC.rpc[rpcProcedureName] = (data, rpcCallback) => {
 					// Provide default parameters if called with non standard parameter, callback
-					rpcCallback =
-						typeof rpcCallback === 'function'
-							? rpcCallback
-							: typeof data === 'function' ? data : () => {};
-					data = data && typeof data !== 'function' ? data : {};
+					if (!data || typeof data === 'function') {
+						data = {};
+					}
+					if (typeof rpcCallback !== 'function') {
+						if (typeof data !== 'function') {
+							rpcCallback = {};
+						}
+						rpcCallback = data;
+					}
```

Comment:
I think data assignment at line 129 is a breaking change and this implementation can be simplified like this:
```suggestion
	if (typeof rpcCallback !== function) {
	  rpcCallback = () => {};
	}
	
	if (typeof data === 'function' ) {
	   rpcCallback = data;
	   data = {};
	}
```